packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts

```diff
@@ -0,0 +1,36 @@
+import { IsInt, IsNotEmpty, IsString } from "class-validator"
+
+import { EntityManager } from "typeorm"
+import { PaymentCollectionService } from "../../../../services"
+
+export default async (req, res) => {
+  const data = req.validatedBody as AdminRefreshPaymentCollectionSessionRequest
+  const { id, session_id } = req.params
+
+  const paymentCollectionService: PaymentCollectionService = req.scope.resolve(
+    "paymentCollectionService"
+  )
+
+  const manager: EntityManager = req.scope.resolve("manager")
+  const payment_collection = await manager.transaction(
+    async (transactionManager) => {
+      return await paymentCollectionService
+        .withTransaction(transactionManager)
+        .refreshPaymentSession(id, session_id, data)
+    }
+  )
+
+  res.status(200).json({ payment_collection })
+}
+
+export class AdminRefreshPaymentCollectionSessionRequest {
+  @IsString()
+  provider_id: string
+
+  @IsString()
+  customer_id: string
+
+  @IsInt()
+  @IsNotEmpty()
+  amount: number
```

Comment:
thought: The store API should not be allowed to specify the amount - should always come from a server-side calculation e.g. based on a cart's contents; let me know if you had another case in mind?

Reply:
That is unnecessary indeed. The refresh will always have the correct amount, I guess I was thinking about having a double check if the user was refreshing the correct session (the amount is validated on the bakcend)